modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/repo/impl/ElasticSearchDetectorRepositoryImpl.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.repo.impl;
+
+import com.expedia.adaptivealerting.modelservice.entity.ElasticSearchDetector;
+import com.expedia.adaptivealerting.modelservice.repo.ElasticSearchDetectorRepoCustom;
+import com.expedia.adaptivealerting.modelservice.util.DateUtil;
+import lombok.val;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.elasticsearch.core.ElasticsearchTemplate;
+import org.springframework.data.elasticsearch.core.query.UpdateQueryBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ElasticSearchDetectorRepositoryImpl implements ElasticSearchDetectorRepoCustom {
+
+    private static final int RESULTS_SIZE = 500;
+
+    @Autowired
+    private ElasticsearchTemplate elasticsearchTemplate;
+
+    @Autowired
+    private Client elasticSearchClient;
+
+    @Override
+    public void toggleDetector(ElasticSearchDetector detector, Boolean enabled) {
```

Comment:
the name of this method looks confusion, it is either adding the detector or updating the detector but saying this method `toggleDetector` does not give any clue about the action.

Reply:
i am just using the existing method names so that we can introduce the change in a non-breaking manner.

Reply:
actually, let me just change the function names since now we are defining an explicit controller and not using the repo function.

Reply:
actually toggle detector sounds like a right word for this. https://stackoverflow.com/questions/2544435/whats-the-word-for-enable-disable